modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/user/common/RESTAPIUserTestBase.java

```diff
@@ -41,4 +51,63 @@ protected void testInit(String apiVersion, String apiDefinition, String tenantDo
         super.init(apiDefinition, basePathInSwagger, basePath);
     }
 
+    /**
+     * @param apiVersion                       api version
+     * @param apiDefinition                    swagger definition of api
+     * @param tenantDomain                     tenant
+     * @param apiUserBasePathInSwagger         base path of endpoint in swagger
+     * @param apiUserBasePathWithTenantContext base path of endpoint with tenant context
+     * @throws XPathExpressionException
+     * @throws RemoteException
+     */
+    protected void testInit(String apiVersion, String apiDefinition, String tenantDomain, String apiUserBasePathInSwagger, String apiUserBasePathWithTenantContext)
+            throws XPathExpressionException, RemoteException {
+
+        String basePathInSwagger = String.format(apiUserBasePathInSwagger, apiVersion);
+        String basePath = String.format(apiUserBasePathWithTenantContext,
+                tenantDomain, apiVersion);
+        super.init(apiDefinition, basePathInSwagger, basePath);
+    }
+
+    protected void initUpdateIDPProperty() throws Exception {
+
+        this.authenticatorClient = new AuthenticatorClient(backendURL);
+        String tenantCookie = this.authenticatorClient.login(ADMIN, ADMIN, isServer.getInstance().getHosts().get("default"));
+        superTenantIDPMgtClient = new IdentityProviderMgtServiceClient(sessionCookie, backendURL);
+        tenantIDPMgtClient = new IdentityProviderMgtServiceClient(tenantCookie, backendURL);
+        superTenantResidentIDP = superTenantIDPMgtClient.getResidentIdP();
+    }
+
+    protected void updateResidentIDPProperty(String propertyKey, String value, boolean
+            isSuperTenant)
+            throws Exception {
+
+        IdentityProviderProperty[] idpProperties = superTenantResidentIDP.getIdpProperties();
+        for (IdentityProviderProperty providerProperty : idpProperties) {
+            if (propertyKey.equalsIgnoreCase(providerProperty.getName())) {
+                providerProperty.setValue(value);
+            }
+        }
+        updateResidentIDP(superTenantResidentIDP, isSuperTenant);
+    }
+
+    private void updateResidentIDP(IdentityProvider residentIdentityProvider, boolean isSuperTenant) throws Exception {
```

Comment:
Add specific exceptions

Reply:
Specific exceptions cannot be added since, methods referred to within this method throws 'Exception' generally.